identity/src/main/java/com/stripe/android/identity/ui/ErrorScreen.kt

```diff
@@ -40,6 +42,17 @@ internal fun ErrorScreen(
     topButton: ErrorScreenButton? = null,
     bottomButton: ErrorScreenButton? = null,
 ) {
+    val cause by identityViewModel.errorCause.observeAsState()
+    LaunchedEffect(cause) {
+        cause?.let {
+            identityViewModel.sendAnalyticsRequest(
+                identityViewModel.identityAnalyticsRequestFactory.genericError(
+                    it.message,
+                    it.stackTraceToString()
+                )
+            )
+        }
+    }
```

Comment:
This seems like it should happen in the view model, not in the UI.

Reply:
I see the tweak, but I’m wondering why any of this is happening in the UI. Can’t we observe `cause` in the view model and send the event there, where it‘s not impacted by the lifecycle of the composable?

Reply:
hmm yeah that's a good idea, actually the reason we have the errorCuase is because it cannot be passed as a parameter when navigting to ErrorScreen, so we ended up posting to this liveData and navigate when error occurs.

Added an observer and removed it in IdentityViewModel to log when it's posted. Note errorCause cannot be removed, as we need to retain it when user cancels the verification